sroka/api/ga/ga.py

```diff
@@ -68,72 +134,131 @@ def print_results(results):
+    :param sampling_level:
+        the GA sampling level (optional, default = HIGHER_PRECISION, valid values: 'DEFAULT', 'FASTER', 'HIGHER_PRECISION')
+
+    :return: a Pandas data frame
+    """
+    if not isinstance(input_dict, Dict):
+        print(f'input_dict={input_dict}')
+        print('input_dict must be a dictionary. You can find valid keys/values in the GA documentation')
         return pd.DataFrame([])
 
-    except HttpError as error:
-        # Handle API errors.
-        print(('Arg, there was an API error : {} : {}'.format(error.resp.status, error._get_reason())))
+    if not isinstance(start_index, int) or start_index < 1:
+        print(f'start_index={start_index}')
+        print('start_index must be a positive integer! The indexing starts from 1 not from 0!')
         return pd.DataFrame([])
 
-    except RefreshError as error:
-        # Handle Auth errors.
-        print('The credentials have been revoked or expired, please re-run '
-              'the application to re-authorize' + str(error))
+    if not isinstance(page_size, int) or page_size < 1:
+        print(f'page_size={page_size}')
+        print('page_size must be an integer. The minimal page size is 1.')
         return pd.DataFrame([])
 
-    except KeyError as error:
-        # Handle wrong or missing values in query.
-        if error.args[0] == 'rows':
-            print('Your query did not return any rows.')
-        else:
-            print('There is an error or missing value in your query: {}'.format(error))
+    if max_pages is not None and (not isinstance(max_pages, int) or max_pages < 1):
+        print(f'max_pages={max_pages}')
+        print('max_pages must be an integer. The minimal value is 1. Use None = retrieve all available pages.')
         return pd.DataFrame([])
 
-    except AssertionError as error:
-        # Handle errors in constructing a query.
-        if not input_dict['dimensions']:
-            print('Your query is missing dimensions.')
-        else:
-            print(('There was an error in constructing your query : {}'.format(error)))
+    if not isinstance(print_sample_size, bool):
+        print(f'print_sample_size={print_sample_size}')
+        print('print_sample_size must be a boolean value')
         return pd.DataFrame([])
+
+    if not isinstance(sampling_level, str) or sampling_level not in ['DEFAULT', 'FASTER', 'HIGHER_PRECISION']:
+        print(f'sampling_level={sampling_level}')
+        print('sampling_level has three valid values: DEFAULT, FASTER, HIGHER_PRECISION')
+        return pd.DataFrame([])
+
+    with __ga_access(input_dict) as service:
+        if 'start_index' in input_dict.keys() or 'max_results' in input_dict.keys():
+            print('This function overwrites start_index and max_results parameters! ' +
+                  'Do not include them in the input_dict parameter.')
+        input_dict = dict(input_dict)
+        input_dict['max_results'] = page_size
```

Comment:
I found that in some cases when page_size is smaller than fetched rows, function throws an error and the result is None. Consider this case:
```
request = {
  'ids': 'ga:89268958',
  'start_date': '2020-03-15',
  'end_date': '2020-03-15',
  'metrics': 'ga:pageviews',
  'filters': 'ga:deviceCategory=~desktop',
  'dimensions': 'ga:deviceCategory,ga:channelGrouping',
  'sort': '-ga:pageviews'
}

df_ga = ga_request_all_data(request, page_size=2)
```